core-web/libs/sdk/experiments/src/lib/persistence/index-db-database-handler.ts

```diff
@@ -0,0 +1,144 @@
+interface DbConfig {
+    db_name: string;
+    db_store: string;
+    db_key_path: string;
+}
+
+const DB_DEFAULT_VERSION = 1;
+
+/**
+ * The `DatabaseHandler` class offers specific methods to store and get data
+ * from IndexedDB.
+ *
+ * @example
+ * // To fetch data from the IndexedDB
+ * const data = await DatabaseHandler.getData();
+ *
+ * @example
+ * // To store an object of type AssignedExperiments to IndexedDB
+ * await DatabaseHandler.persistData(anAssignedExperiment);
+ *
+ * @example
+ * // To get an object of type AssignedExperiments to IndexedDB
+ * await DatabaseHandler.persistData(anAssignedExperiment);
+ *
+ */
+
+export class IndexDBDatabaseHandler {
+    constructor(private config: DbConfig) {
+        if (!config) {
+            throw new Error('Config is required');
+        }
+
+        const { db_name, db_store, db_key_path } = config;
+
+        if (!db_name) {
+            throw new Error("'db_name' is required in config");
+        }
+
+        if (!db_store) {
+            throw new Error("'db_store' is required in config");
+        }
+
+        if (!db_key_path) {
+            throw new Error("'db_key_path' is required in config");
+        }
+    }
+
+    /**
+     * Saves the provided data to indexDB.
+     *
+     * @async
+     * @param {AssignedExperiments} data - The data to be saved.
+     * @returns {Promise<any>} - The result of the save operation.
+     */
+    public async persistData<T>(data: T): Promise<IDBValidKey> {
+        const db = await this.openDB();
+
+        return await new Promise((resolve, reject) => {
+            const transaction = db.transaction([this.config.db_store], 'readwrite');
+            const store = transaction.objectStore(this.config.db_store);
+            const request = store.put(data, this.config.db_key_path);
+
+            request.onsuccess = () => resolve(request.result);
+            request.onerror = () => reject(request.error);
+        });
```

Comment:
This code is repeated in the method below with the only change that the transaction is readonly and there's no being put.

Can we abstract this to a method and just pass the `readwrite/readonly` and whatever we need as a parameters?